docs/fsharp/tutorials/asynchronous-and-concurrent-programming/async.md

```diff
@@ -1,213 +1,349 @@
+```
 
-There are a few syntactical constructs which are worth noting:
+In the example, the `countFileBytes` function is of type `string -> Async<unit>`. Calling the function does not actually execute the asynchronous computation. Instead, it returns an `Async<unit>` that acts as a _specification_ of the work that is to execute asynchronously.
 
-*   `let!` binds the result of an async expression (which runs on another context).
-*   `use!` works just like `let!`, but disposes its bound resources when it goes out of scope.
-*   `do!` will await an async workflow which doesn’t return anything.
-*   `return` simply returns a result from an async expression.
-*   `return!` executes another async workflow and returns its return value as a result.
+The `Async.Start` function is used to start the computation.
 
-Additionally, normal `let`, `use`, and `do` keywords can be used alongside the async versions just as they would in a normal function.
+Additionally, `Async.AwaitTask` is called in the `countFileBytes` function. This creates an `Async<'T>` that runs <xref:System.IO.File.WriteAllBytesAsync%2A> (which returns a `Task<byte[]>`) and returns its result.
 
-## How to start Async Code in F# #
+This is a key difference with the C#/VB style of `async` programming. In F#, asynchronous computations can be thought of as **Cold tasks**. They must be explicitly started to actually execute. This has some advantages, as it allows you to combine and sequence asynchronous work easily.
 
-As mentioned earlier, async code is a specification of work to be done in another context which needs to be explicitly started. Here are two primary ways to accomplish this:
+## Combining asynchronous computations
 
-1.  `Async.RunSynchronously` will start an async workflow on another thread and await its result.
+Here is an example that builds upon the previous one by combining computations:
 
 ```fsharp
 open System
-open System.Net
+open System.IO
 
-let fetchHtmlAsync url = 
+let countFileBytes path =
     async {
-        let uri = Uri(url)
-        use webClient = new WebClient()
-        let! html = webClient.AsyncDownloadString(uri)
-        return html
+        let! bytes = File.ReadAllBytesAsync(path) |> Async.AwaitTask
+        let fileName = Path.GetFileName(path)
+        printfn "File %s has %d bytes" fileName bytes.Length
     }
 
- // Execution will pause until fetchHtmlAsync finishes
- let html = "https://dotnetfoundation.org" |> fetchHtmlAsync |> Async.RunSynchronously
+[<EntryPoint>]
+let main argv =
+    argv
+    |> Array.map countFileBytes
+    |> Async.Parallel
+    |> Async.Ignore
+    |> Async.Start
 
- // you actually have the result from fetchHtmlAsync now!
- printfn "%s" html
- ```
+    Console.Read() |> ignore
+    0 // return an integer exit code
+```
+
+As you can see, the `main` function has quite a few more calls made. Conceptually, it does the following:
+
+1. Transform the command-line arguments into `Async<unit>` computations with `Array.map`
```

Comment:
computaiton builders? workflows ? I don't like the term computations for Async<T>, though in fairness the term is applied consistently

Reply:
I'm going with computation so that there is consistent terminology with CEs in our documentation. But I'll have a "terminology" section at the top that mentions other, often equivalent terms.